features/pre-post-transformation/README.md

```diff
@@ -0,0 +1,119 @@
+| pre           | string           | -                                  | Request parameter for Pre Transformation                                                                                                                                  |
+| post          | array of objects | -                                  | Request parameter for Post Transformation                                                                                                                                 |
+| post.type     | string           | Yes                                | Type of required post-transformation.<br/>Allowed Values: `abs`, `thumbnail`, `transformation`, `gif-to-video`                                                            |
+| post.value    | string           | If `post.type` is `transformation` | Value of required post-transformation.                                                                                                                                    |
+| post.protocol | string           | If `post.type` is `abs`            | ABS Streaming Protocol. <br/> Allowed Values: `hls`, `dash`<br/>See [Adaptive bitrate streaming](../video-transformation/adaptive-bitrate-streaming.md) for more details. |
+
+Example of a pre transformation:
+
+```javascript
+// Request Body (multipart/form-data) for Upload API
+{
+  /*
+  ...rest of the Upload API request parameters
+  */
+  "transformation": {
+    "pre": "rt-90"
+  }
+}
+```
+
+Example of a request with multiple post transformations:
+
+```javascript
+// Request Body (multipart/form-data) for Upload API
+{
+  /*
+  ...rest of the Upload API request parameters
+  */
+  "transformation": {
+    "post": [
+        {
+          "type": "abs",
+          "value: "sr-240_360_480_720_1080",
+          "protocol": "dash"
+        },
+        {
+          "type": "transformation",
+          "value": "h-300"
+        }
+    ]
+  }
+}
+```
+
+You can also use pre & post transformations together in a single request. For an example, we have a request with a pre-transformation to change a video's [width](../video-transformation/resize-crop-and-other-common-video-transformations.md#width---w) to "500px" & then eagerly generate a thumbnail for it using a post-transformation.
+
+```javascript
+// Request Body (multipart/form-data) for Upload API
+{
+  /*
+  ...rest of the Upload API request parameters
+  */
+  "transformation": {
+    "pre": "w-500",
+    "post": [
+      {
+        "type": "thumbnail"
+      }
+    ]
+  }
+}
+```
+
+> Note: Don't forget to stringify the contents of the `post` property before making a request.
+
+## Asynchronous behaviour
+
+- Post transformations are always async.
+- Pre transformation is sync for image files but async for video files.
+
```

Comment:
Mention that use webhooks to build your workflow. Different sections in the docs should be connected ensuring a flow.

Reply:
Done in https://github.com/imagekitio/docs-keep-a-changelog/pull/201/commits/376303b5a5e6c3e690d5cf118c9b131ea186fb65